stream/src/main/java/org/apache/apex/malhar/stream/window/impl/DefaultWindowedStorageImpl.java

```diff
@@ -0,0 +1,120 @@
+package org.apache.apex.malhar.stream.window.impl;
+
+import com.datatorrent.api.StreamCodec;
+import org.apache.apex.malhar.stream.window.Window;
+import org.apache.apex.malhar.stream.window.WindowedStorage;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.TreeSet;
+
+/**
+ * Created by david on 6/14/16.
+ */
+public class DefaultWindowedStorageImpl<K, V> implements WindowedStorage<K, V>
```

Comment:
In the spirit of designing an interface that allows for efficient storage operation with large state, can you consider 
https://issues.apache.org/jira/browse/APEXMALHAR-2026
as well as managed state?